base/shared/src/main/scala/scalaz/data/alist1.scala

```diff
@@ -27,8 +27,11 @@ sealed abstract class AList1[F[_, _], A, B] {
   def ::[Z](fza: F[Z, A]): AList1[F, Z, B] =
     ACons1(fza, this.toList)
 
-  def :+[C](f: F[B, C]): AList1[F, A, C] =
-    this ::: AList1(f)
+  def :+[Z](fza: F[Z, A]): AList1[F, Z, B] =
+    ::(fza)
```

Comment:
I think the original meaning of `:+` was the right one, wasn't it?

Reply:
The tests say otherwise, from what I can tell.

Reply:
Then the tests must be wrong?

Reply:
I suggest explicitly writing out `fza :: this`. `:+` usually means "append on the right" and it should be equivalent to `andThen lift(x)`.

Reply:
Check the tests, they seem right to me.

Reply:
To be clear then, this is correct?
```scala
        test("foldBalanced") { () =>
          final class Fake[A, B](val str: String)
          // not lawful, used to observe balanced binary tree shape of `foldBalanced`.
          // not associative.
          implicit val fakeSemicategory: Semicategory[Fake] = instanceOf(new SemicategoryClass[Fake] {
            def compose[A, B, C](fst: Fake[B, C], snd: Fake[A, B]): Fake[A, C] =
              new Fake("(" + fst.str + "|" + snd.str + ")")
          })
          def fake(str: String): Fake[Int, Int] = new Fake(str)
          val result = "((bar|foo)|(world|hello))"
          IList(
            (lift(fake("hello")) :+ fake("world") :+ fake("foo") :+ fake("bar")).foldBalanced.str,
            (fake("hello") +: fake("world") +: fake("foo") +: lift(fake("bar"))).foldBalanced.str
          ).foldMap(assertEqual(_, result))
        }
```

Reply:
Agreed.